src/main/java/com/github/manolo8/darkbot/core/utils/factory/EntityFactory.java

```diff
@@ -17,39 +19,68 @@
 
 import static com.github.manolo8.darkbot.Main.API;
 
-//Maybe split them same as classes in darkorbit?
 public enum EntityFactory {
-    UNKNOWN     (Entity::new),
-    //Ship and Npc have same class
-    SHIP        (Ship::new),
-    NPC         (Npc::new),
-    PORTAL      ("[0-9]+$"),
-    BOX         (Box::new,           "box_.*"),
-    ORE         (Box::new,           "ore_.*"),
-    MINE        (Box::new,           "mine_.*"),
-    NPC_BEACON  (MapNpc::new,        "npc-beacon.*"),
-    LOW_RELAY   (MapNpc::new,        "relay"),
-    //Barrier and Mist zone have same class
-    BARRIER     (Barrier::new,       "NOA|DMG"),
-    MIST_ZONE   (NoCloack::new),
-    WRECK_MODULE(BattleStation::new, "wreck"),
-    ASTEROID    (BattleStation::new, "asteroid|cbs-construction"),
-    MODULE      (BattleStation::new, "module_.*|module-construction"),
-    STATION     (BattleStation::new, "battleStation"),
-    BASE_POINT  (BasePoint::new,     "(questgiver|repairstation|headquarters|refinery|hangar|turret|station)_.*");
+    BOX      (Box::new,       "box_.*"),
+    ORE      (Box::new,       "ore_.*"),
+    MINE     (Box::new,       "mine_.*"),
+    FIREWORK (Entity::new,    "firework_box"),
+    X2_BEACON(BasePoint::new, "beacon_.*"),
+
+    LOW_RELAY (MapNpc::new, "relay"),
+    NPC_BEACON(MapNpc::new, "npc-beacon.*"),
+
+    CBS_ASTEROID    (BattleStation::new, "asteroid"),
+    CBS_CONSTRUCTION(BattleStation::new, "cbs-construction"),
+    CBS_MODULE      (BattleStation::new, "module_.*"),
+    CBS_MODULE_CON  (BattleStation::new, "module-construction"),
+    CBS_WRECK_MODULE(BattleStation::new, "wreck"),
+    CBS_STATION     (BattleStation::new, "battleStation"),
+
+    POD_HEAL        (BasePoint::new, "pod_heal"),
+    SPACE_BALL      (BasePoint::new, "mapIcon_spaceball"),
+    BUFF_CAPSULE    (BasePoint::new, "buffCapsule_.*"),
+    BURNING_TRAIL   (BasePoint::new, "burning_trail_entity_.*"),
+    PLUTUS_GENERATOR(BasePoint::new, "plutus-generator"),
+
+    REFINERY      (BasePoint::new, "refinery_.*"),
+    HOME_ZONE     (BasePoint::new, "ctbHomeZone_.*"),
+    BASE_TURRET   (BasePoint::new, "turret_.*"),
+    BASE_HANGAR   (BasePoint::new, "hangar_.*"),
+    BASE_STATION  (BasePoint::new, "station_.*"),
+    HEADQUARTER   (BasePoint::new, "headquarters_.*"),
+    QUEST_GIVER   (BasePoint::new, "questgiver_.*"),
+    REPAIR_STATION(BasePoint::new, "repairstation_.*"),
+
+    BARRIER  (Barrier::new, EntityFactory::defineZoneType, "NOA|DMG"),
+    MIST_ZONE(NoCloack::new),
+
+    SHIP(Ship::new, EntityFactory::defineShipType),
+    NPC (Npc::new),
+
+    PET    (Pet::new),
+    PORTAL ("[0-9]+$"),
```

Comment:
```suggestion
    PORTAL (EntityFactory::getOrCreatePortal, "[0-9]+$"),
```
Make this be the call that creates a portal, so that you can get rid of specific logic that does it (see below)